src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -0,0 +1,1077 @@
+			});
+		this.editorInput = this.instantiationService.createInstance(WalkThroughInput, {
+			typeId: welcomeInputTypeId,
+			name: localize('welcome.title', "Welcome"),
+			resource,
+			telemetryFrom,
+			onReady: (container: HTMLElement) => this.onReady(container, recentlyOpened, installedExtensions, fileService)
+		});
+	}
+	public openEditor() {
+		return this.editorService.openEditor(this.editorInput, { pinned: false });
+	}
+	private onReady(container: HTMLElement, recentlyOpened: Promise<IRecentlyOpened>, installedExtensions: Promise<IExtensionStatus[]>, fileService: IFileService): void {
+		const enabled = isWelcomePageEnabled(this.configurationService, this.contextService);
+		const showOnStartup = <HTMLInputElement>container.querySelector('#showOnStartup');
+		if (enabled) {
+			showOnStartup.setAttribute('checked', 'checked');
+		}
+		showOnStartup.addEventListener('click', e => {
+			this.configurationService.updateValue(configurationKey, showOnStartup.checked ? 'welcomePage' : 'newUntitledFile', ConfigurationTarget.USER);
+		});
+		const prodName = container.querySelector('.welcomePage .title .caption') as HTMLElement;
+		if (prodName) {
+			prodName.innerHTML = this.productService.nameLong;
+		}
+		recentlyOpened.then(async ({ workspaces }) => {
+			// Filter out the current workspace
+			workspaces = workspaces.filter(recent => !this.contextService.isCurrentWorkspace(isRecentWorkspace(recent) ? recent.workspace : recent.folderUri));
+			if (!workspaces.length) {
+				const recent = container.querySelector('.welcomePage') as HTMLElement;
+				recent.classList.add('emptyRecent');
+				return;
+			}
+			const ul = container.querySelector('.recent ul');
+			if (!ul) {
+				return;
+			}
+			const workspacesToShow = workspaces.slice(0, 5);
+			const updateEntries = async () => {
+				while (ul.firstChild) {
+					ul.removeChild(ul.firstChild);
+				}
+				await this.mapListEntries(workspacesToShow, fileService);
+			};
+			await updateEntries();
+			this._register(this.labelService.onDidChangeFormatters(updateEntries));
+		}).then(undefined, onUnexpectedError);
+		this.addExtensionList(container, '.extension_list');
+		this.addExtensionPack(container, '.extensionPack');
+		this.updateInstalledExtensions(container, installedExtensions);
+		this._register(this.instantiationService.invokeFunction(onExtensionChanged)(ids => {
+			for (const id of ids) {
+				if (container.querySelector(`.installExtension[data-extension="${id.id}"], .enabledExtension[data-extension="${id.id}"]`)) {
+					const installedExtensions = this.instantiationService.invokeFunction(getInstalledExtensions);
+					this.updateInstalledExtensions(container, installedExtensions);
+					break;
+				}
+			}
+		}));
+		this.createDropDown();
+		this.createDesktopPreviewToolTip();
+		this.createPreviewModal();
+	}
+
+	private createDesktopPreviewToolTip() {
+		const previewLink = document.querySelector('#preview_link--desktop');
```

Comment:
I see a lot of desktop/mobile references. Are there any plans for this to actually be on mobile? It seems like we'd need to do a whole lot more changes to the core experience to make that a usable experience.

Reply:
resolved